src/patternfly/components/List/list.hbs

```diff
@@ -1,4 +1,7 @@
 <{{#if list--type}}{{list--type}}{{else}}ul{{/if}} class="pf-c-list{{#if list--modifier}} {{list--modifier}}{{/if}}"
+  {{#unless list--type}}
```

Comment:
Datalist we'll want to keep the role on since it'll run into the same bug in Safari + VO.

Good catch on the `ol` in List, will update that logic

Reply:
I removed the `unless` block from List since that component should really only be an `ol` or `ul`. I can't imagine an example being added where `dl` is passed.